05_stringer/nickolee/stringer_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := "127.0.0.1"
+	actual := buf.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestStringMethod(t *testing.T) {
+	type test struct {
+		name     string
+		input    IPAddr
+		expected string
+	}
+
+	tests := []test{
```

Comment:
Nice work. Since you are also only ever using this struct as a one time thing, you can avoid declaring it as a type

```go
tests := struct{
  name string
  input IPAddr
  expected string
}{
  {name: "all local IPv4 addresses", input: IPAddr{0, 0, 0, 0}, expected: "0.0.0.0"},
}
```